component/src/main/java/org/ballerinalang/net/jms/actions/Poll.java

```diff
@@ -70,33 +67,30 @@ public ConnectorFuture execute(Context context) {
 
         // Get the map of properties.
         BStruct  connectorConfig = ((BStruct) bConnector.getRefField(0));
+        String acknowledgementMode = connectorConfig.getStringField(4);
 
-        Map<String, String> propertyMap = JMSUtils.preProcessJmsConfig(connectorConfig);
+        // Retrieve transport client
+        JMSClientConnector jmsClientConnector = (JMSClientConnector) bConnector
+                .getnativeData(Constants.JMS_TRANSPORT_CLIENT_CONNECTOR);
 
         // Get the connector key
         String connectorKey = bConnector.getStringField(0);
 
-        propertyMap.put(JMSConstants.PARAM_DESTINATION_NAME, destination);
-
-        boolean isTransacted = Boolean.FALSE;
-        if (propertyMap.get(JMSConstants.PARAM_ACK_MODE) != null) {
-            isTransacted = (JMSConstants.SESSION_TRANSACTED_MODE.equals(propertyMap.get(JMSConstants.PARAM_ACK_MODE))
-                    || JMSConstants.XA_TRANSACTED_MODE.equals(propertyMap.get(JMSConstants.PARAM_ACK_MODE))) && context
-                    .isInTransaction();
-        }
-
         try {
-            JMSClientConnector jmsClientConnector = new JMSConnectorFactoryImpl().createClientConnector(propertyMap);
-            if (log.isDebugEnabled()) {
```

Comment:
don't we need this log in the init action?

Reply:
I have added it